authors/apps/authentication/serializers.py

```diff
@@ -28,6 +32,30 @@ class Meta:
 
     def create(self, validated_data):
         # Use the `create_user` method we wrote earlier to create a new user.
+        password = validated_data.get('password') #we get a key from the dictionary validated data
+        email = validated_data.get('email')
+        username = validated_data.get('username')
+        pattern = re.compile(
+                r"^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,128}$"
+            )
+        if not pattern.match(password):
+            raise serializers.ValidationError(
+                {
+                    "password" :[
+                    "password must contain a numeric character"
+                    ]
+                }
+                )
+
+        
+        elif len(username)< 4 or not username.isalnum():
```

Comment:
Please slack Trust regarding whether users can be allowed to add a character such as an underscore or a dash in their username. This .isalnum() method restricts their use but I believe they should be used. The PO is to have a final say regarding this.

Reply:
The PO says we should allow only numbers and letters